tests/ut/ir/transforms/test_expand_mixed_kernel_split_aiv.py

```
@@ -392,6 +392,128 @@ def test_folded_transport_stays_on_the_box_partition_without_a_stride():
+    )
+    return ir.Program([func], "test_cube_produced_gather", span)
+
+
+def test_shard_of_a_vector_produced_value_is_rejected():
+    """A vector-produced shard operand is an authoring error, not a lane-local halve.
+
+    ``pl.aiv_shard`` MEANS "cross the AIC/AIV boundary"; a value the AIV lane
+    already produced has no crossing to name. Reject it with the boundary op's
+    span rather than lowering it into a tpush the cube lane cannot satisfy.
+    """
+    with pytest.raises(ValueError, match="is produced on the VECTOR lane by 'tile.full'"):
```

Ruff (0.16.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)